tensorboard/webapp/plugins/text_v2/data_source/text_v2_types.ts

```diff
@@ -0,0 +1,42 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs';
+
+export interface BackendRunToTagsMap {
+  [runName: string]: string[];
+}
+
+export interface BackendStepDatum {
```

Comment:
It would be less confusing if we don't mix types for the TextDataSource interface (BackendRunToTagsMap, StepDatum) with types that are internal to the implementation (BackendStepDatum). Consider move this particular interface into the implementation class, where you don't have to `export` it, or to its own file?

Reply:
Refactored to make backend types contained in the backend specific data provider but instead have only frontend types exported. Thanks for pointing that out.